### ISSUE TITLE
docfx metadata hangs at: Determining projects to restore...  Nothing to do. None of the projects specified contain packages to restore.

### ISSUE THREAD

**Issue by bartofzo** [dotnet/docfx#10144]
**Describe the bug**
All of the sudden, when running docfx metadata, I get the message:

Loading project /Users/bart/Develop/anypath/AnyPath/AnyPath.csproj
  Determining projects to restore...
  Nothing to do. None of the projects specified contain packages to restore.

And nothing happens. 

**Expected behavior**
Metadata for the project should be built but instead nothing happens.

**Context (please complete the following information):**
- OS: MacOS  Ventura 13.2.1
- Docfx version: 2.77.0

**Additional context**
This is a Unity project with Rider generating the csproj files.
This used to always just work, but all of the sudden it doesn't anymore and I have no
idea what changed.

**Comment by adrian-moll** [comment]
We had the same issue when building the documentation in our AzureDevops pipeline.
The problem on our side seemed to be that we altered the buildNumber during the run (using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning)).

I do not really understand how this could interfere with docFx but when we disabled this feature in Nerdbank.GitVersioning the docFx command completed successfully.

Here the configuration in the version.json of Nerdbank that triggered the issue:

```
"cloudBuild": {
  "setVersionVariables": false, 
  "buildNumber": {
    "enabled": true // When true --> docFx "hangs" and doesn't complete
  }
```

**Comment by goto01** [comment]
Hello, @bartofzo. I'm encountering the same issue with **brew** and **dotnet**. You can try downloading the DocFX build directly from the [releases](https://github.com/dotnet/docfx/releases). However, have you found a solution for this issue?

**Comment by bartofzo** [comment]
> Hello, @bartofzo. I'm encountering the same issue with **brew** and **dotnet**. You can try downloading the DocFX build directly from the [releases](https://github.com/dotnet/docfx/releases). However, have you found a solution for this issue?

I don't know how to install docfx that way. I've already updated it to the latest release using
`dotnet tool update -g docfx`
which says it reinstalled the latest stable version 2.77.0

Is there a difference? 

No, I haven't resolved it yet unfortunately...

**Comment by bartofzo** [comment]
> We had the same issue when building the documentation in our AzureDevops pipeline. The problem on our side seemed to be that we altered the buildNumber during the run (using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning)).
> 
> I do not really understand how this could interfere with docFx but when we disabled this feature in Nerdbank.GitVersioning the docFx command completed successfully.
> 
> Here the configuration in the version.json of Nerdbank that triggered the issue:
> 
> ```
> "cloudBuild": {
>   "setVersionVariables": false, 
>   "buildNumber": {
>     "enabled": true // When true --> docFx "hangs" and doesn't complete
>   }
> ```

I'm not using NerdBank.GitVersioning. Where does it modify the build number? Is there something on my project side I shoud look into? (I've never modified anything about my project except for code and this issue just appeared out of the blue by the way)

**Comment by adrian-moll** [comment]
> > We had the same issue when building the documentation in our AzureDevops pipeline. The problem on our side seemed to be that we altered the buildNumber during the run (using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning)).
> > I do not really understand how this could interfere with docFx but when we disabled this feature in Nerdbank.GitVersioning the docFx command completed successfully.
> > Here the configuration in the version.json of Nerdbank that triggered the issue:
> > ```
> > "cloudBuild": {
> >   "setVersionVariables": false, 
> >   "buildNumber": {
> >     "enabled": true // When true --> docFx "hangs" and doesn't complete
> >   }
> > ```
> 
> I'm not using NerdBank.GitVersioning. Where does it modify the build number? Is there something on my project side I shoud look into? (I've never modified anything about my project except for code and this issue just appeared out of the blue by the way)

NerdBank.GitVersionning affects the build number of the assembly (incrementing the version based on git-history) and additionally sets some environment variables (influencing the BuildNumber inside our AzureDevops Pipeline) --> unfortunately i dont know what exactly what it does behind the scenes.

When I run docfx manually on my machine everyhting works. We only have the issue inside our build pipeline (and only when enabling the buildNumber options).

Do you have any special tool that affects your build ?

**Comment by goto01** [comment]
> > Hello, @bartofzo. I'm encountering the same issue with **brew** and **dotnet**. You can try downloading the DocFX build directly from the [releases](https://github.com/dotnet/docfx/releases). However, have you found a solution for this issue?
> 
> I don't know how to install docfx that way. I've already updated it to the latest release using `dotnet tool update -g docfx` which says it reinstalled the latest stable version 2.77.0
> 
> Is there a difference?
> 
> No, I haven't resolved it yet unfortunately...

You can download the compiled build from the Releases section [here](https://github.com/dotnet/docfx/releases).

In the assets section download archive for your system and unzip it.

<img width="977" alt="Screenshot 2024-08-21 at 10 46 07" src="https://github.com/user-attachments/assets/5cb897a7-5803-46bd-adb5-a644270d276a">

Currently i'm using [v2.75.3](https://github.com/dotnet/docfx/releases/tag/v2.75.3)

**Comment by bartofzo** [comment]
> > > Hello, @bartofzo. I'm encountering the same issue with **brew** and **dotnet**. You can try downloading the DocFX build directly from the [releases](https://github.com/dotnet/docfx/releases). However, have you found a solution for this issue?
> > 
> > 
> > I don't know how to install docfx that way. I've already updated it to the latest release using `dotnet tool update -g docfx` which says it reinstalled the latest stable version 2.77.0
> > Is there a difference?
> > No, I haven't resolved it yet unfortunately...
> 
> You can download the compiled build from the Releases section [here](https://github.com/dotnet/docfx/releases).
> 
> In the assets section download archive for your system and unzip it.
> 
> <img alt="Screenshot 2024-08-21 at 10 46 07" width="977" src="https://private-user-images.githubusercontent.com/8189011/359822949-5cb897a7-5803-46bd-adb5-a644270d276a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMzA0MzEsIm5iZiI6MTcyNDIzMDEzMSwicGF0aCI6Ii84MTg5MDExLzM1OTgyMjk0OS01Y2I4OTdhNy01ODAzLTQ2YmQtYWRiNS1hNjQ0MjcwZDI3NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjFUMDg0ODUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzMxYjY5YzUxYTVlNzEwZGVkNjc3MWYxYmIwNjAxYTZhMmY1OTMyN2MyYjZjZDRjYTMzNmEzMDg1ZDM4MDU0MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XBY1TvJvDZNTonVdnR34Wgh7y2oGw8sgLSqTyuQL4xU">
> Currently i'm using [v2.75.3](https://github.com/dotnet/docfx/releases/tag/v2.75.3)

Yes, but how do I install it? The documentation only shows installation via the command line.

**Comment by goto01** [comment]
Just download the archive and use the **DocFX** file within to generate documentation. For example, I downloaded the archive for macOS and unzipped it into `/Users/my-user-name/Documents/Docfx`.

In this folder, you can use the DocFX file to run DocFX in your project folder like this:
`/Users/my-user-name/Documents/Docfx/docfx docfx-config.json --serve`

**Comment by bartofzo** [comment]
> Just download the archive and use the **DocFX** file within to generate documentation. For example, I downloaded the archive for macOS and unzipped it into `/Users/my-user-name/Documents/Docfx`.
> 
> In your project folder, you can use the DocFX file to run DocFX in your project folder like this: `/Users/my-user-name/Documents/Docfx/docfx docfx-config.json --serve`

Ok, I did this (also had to use xattr to make it run because of the malware protection).
No difference... still hangs at Nothing to restore.

**Comment by bartofzo** [comment]
> > > We had the same issue when building the documentation in our AzureDevops pipeline. The problem on our side seemed to be that we altered the buildNumber during the run (using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning)).
> > > I do not really understand how this could interfere with docFx but when we disabled this feature in Nerdbank.GitVersioning the docFx command completed successfully.
> > > Here the configuration in the version.json of Nerdbank that triggered the issue:
> > > ```
> > > "cloudBuild": {
> > >   "setVersionVariables": false, 
> > >   "buildNumber": {
> > >     "enabled": true // When true --> docFx "hangs" and doesn't complete
> > >   }
> > > ```
> > 
> > 
> > I'm not using NerdBank.GitVersioning. Where does it modify the build number? Is there something on my project side I shoud look into? (I've never modified anything about my project except for code and this issue just appeared out of the blue by the way)
> 
> NerdBank.GitVersionning affects the build number of the assembly (incrementing the version based on git-history) and additionally sets some environment variables (influencing the BuildNumber inside our AzureDevops Pipeline) --> unfortunately i dont know what exactly what it does behind the scenes.
> 
> When I run docfx manually on my machine everyhting works. We only have the issue inside our build pipeline (and only when enabling the buildNumber options).
> 
> Do you have any special tool that affects your build ?

I don't think so. It's a Unity project. The solution is automatically generated.

**Comment by bartofzo** [comment]
> > > Hello, @bartofzo. I'm encountering the same issue with **brew** and **dotnet**. You can try downloading the DocFX build directly from the [releases](https://github.com/dotnet/docfx/releases). However, have you found a solution for this issue?
> > 
> > 
> > I don't know how to install docfx that way. I've already updated it to the latest release using `dotnet tool update -g docfx` which says it reinstalled the latest stable version 2.77.0
> > Is there a difference?
> > No, I haven't resolved it yet unfortunately...
> 
> You can download the compiled build from the Releases section [here](https://github.com/dotnet/docfx/releases).
> 
> In the assets section download archive for your system and unzip it.
> 
> <img alt="Screenshot 2024-08-21 at 10 46 07" width="977" src="https://private-user-images.githubusercontent.com/8189011/359822949-5cb897a7-5803-46bd-adb5-a644270d276a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyNTQxNDEsIm5iZiI6MTcyNDI1Mzg0MSwicGF0aCI6Ii84MTg5MDExLzM1OTgyMjk0OS01Y2I4OTdhNy01ODAzLTQ2YmQtYWRiNS1hNjQ0MjcwZDI3NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjFUMTUyNDAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTBjOTU4ZmVjODQ0ODEyMDAxMzgzMWZiZGRmZTVmMDY2MDMzMmRiMmJkZmE5NTNmZDUwYmE4MDFlNmY4M2IzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-3GkMNHovXJY0rQNH1jb_W0hcsXeFI-SuEvECEQvE_4">
> Currently i'm using [v2.75.3](https://github.com/dotnet/docfx/releases/tag/v2.75.3)

Ok so I've tried v2.75.3 and this opens up another can of worms. Running this gives me the following error:

Failed to find all versions of .NET Core MSBuild. Call
`to hostfxr_resolve_sdk2. There may be more details in stderr.`

Which I've encountered before. I used to be able to resolve this by running:
`export PATH=/opt/homebrew/Cellar/dotnet/7.0.100/libexec:$PATH`

But apparently, I don't have dotnet 7 anymore and it is now dotnet 8.
If I change the path to where dotnet8 .0.4 is located (don't ask me while this is even necessary for docfx to run...)

Then I get the following error:
`BadImageFormatException: Could not load file or assembly 'Microsoft.Build.Framework, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An attempt was made to load a program with an incorrect format.`

This is all of course with version v2.75.3 of docfx which is not the latest release.
Running the most recent version of docfx still gives me the "Nothing to do" hang.

Are there any (Apple silicon) mac users out there that are able to run docfx without any issues?

**Comment by goto01** [comment]
The whole situation with trying to run docfx on my mac is really odd. I have problems getting it to work, and the only way I’ve been able to run it on macbook m1 is by using the archive from the release section; all other options don’t work properly. I also have dotnet 8 installed.

**Comment by bartofzo** [comment]
Managed to work around it by building my project using Rider and running docfx metadata directly on the .dll file with the latest version.

**Comment by ptupitsyn** [comment]
Got the same issue after .NET 6 -> .NET 8 upgrade: docfx hangs, and only on CI (works locally).

`Nerdbank.GitVersioning` is the problem:
* Removing it from the project fixes the build
* Disabling **both** `setVersionVariables` and `buildNumber.enabled` fixes the build

Resulting `version.json` - **docfx works**:
```json
{
    "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/main/src/NerdBank.GitVersioning/version.schema.json",
    "version": "3.0.0-snapshot",
    "publicReleaseRefSpec": [
        "^refs/heads/main",
        "^refs/heads/\\d+.*$"
    ],
    "cloudBuild": {
        "setVersionVariables": false,
        "buildNumber": {
            "enabled": false
        }
    }
}
```
